### PR TITLE
Resend confirmation email on lite user registration if the lite user already existing

### PR DESCRIPTION
**Pull request by bhagyasakalanka**
### Proposed changes in this pull request

- Resend confirmation email on lite user registration if the lite user already existing
- Introduce reacaptcha connector for lite user registration

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/connector/LiteRegistrationConfigImpl.java [hunk 1]
```diff
@@ -100,6 +100,9 @@ public Map<String, String> getPropertyNameMapping() {
         nameMapping.put(IdentityRecoveryConstants.ConnectorConfig.LITE_REGISTRATION_CALLBACK_REGEX,
                 "Lite user registration callback URL regex");
         nameMapping.put(LIST_PURPOSE_PROPERTY_KEY, "Manage Lite-Sign-Up purposes");
+        nameMapping.put(
+                IdentityRecoveryConstants.ConnectorConfig.LITE_REGISTRATION_RESEND_VERIFICATION_ON_USER_EXISTENCE,
+                "Resend confirmation email on lite user existence");
```
Comment (bhagyasakalanka) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/connector/LiteRegistrationConfigImpl.java:105]: Change the message.

Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/connector/LiteRegistrationConfigImpl.java]: Fixed

File: components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/LiteApiServiceImpl.java [hunk 2]
```diff
@@ -90,7 +98,26 @@ public Response litePost(LiteUserRegistrationRequestDTO liteUserRegistrationRequ
                 LOG.debug("Client Error while self registering lite user ", e);
             }
             if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_USER_ALREADY_EXISTS.getCode().equals(e.getErrorCode())) {
-                Utils.handleConflict(e.getMessage(), e.getErrorCode());
+                try {
+                    boolean isResendEnabledOnUserExistence =
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/LiteApiServiceImpl.java:102]: Shall we change the variable name to isResendCodeEnabledOnUserExistence/isResendVerificationEnabledOnUserExistence?

Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/LiteApiServiceImpl.java]: Changed to `isResendVerificationEnabledOnUserExistence`

**Review by nilasini — APPROVED** [review]

**Pull request merged**